### PR TITLE
feat(core): update connector db schema

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

1. Update connector DB schema:
Add `id` as the primary key.
Create an index that contains `platform` and `name` (connector name) and can find the only connector.
2. Fix UT accordingly.

## Linear Issue Reference

LOG-2309
LOG-2097

## Testing

Passed UTs.

### CODE REVIEW

File: packages/connector-types/src/index.ts [hunk 1]
```diff
@@ -1,21 +1,34 @@
...
-  type: ConnectorType;
-  name: Record<Languages, string>;
-  logo: string;
-  description: Record<Languages, string>;
-  readme: string;
-  configTemplate: string;
+export enum ConnectorPlatform {
+  General = 'General',
+  Native = 'Native',
+  NA = 'NotApplied',
```
Comment (simeng-li) [inline packages/connector-types/src/index.ts:13]: what does NotApplied stand for?   Can it be avoided?

File: packages/connector-types/src/index.ts [hunk 2]
```diff
@@ -1,21 +1,34 @@
...
-export interface ConnectorMetadata {
-  id: string;
-  type: ConnectorType;
-  name: Record<Languages, string>;
-  logo: string;
-  description: Record<Languages, string>;
-  readme: string;
-  configTemplate: string;
+export enum ConnectorPlatform {
+  General = 'General',
```
Comment (simeng-li) [inline packages/connector-types/src/index.ts:11]: nit: maybe univeral?

File: packages/core/src/connectors/index.ts [hunk 3]
```diff
@@ -108,11 +109,24 @@ export const getConnectorInstanceByType = async <T extends ConnectorInstance>(
 
 export const initConnectors = async () => {
   const connectors = await findAllConnectors();
-  const existingConnectors = new Map(connectors.map((connector) => [connector.id, connector]));
+  const existingConnectors = new Map(
+    connectors.map((connector) => [[connector.name, connector.platform].join(' '), connector])
+  );
   const newConnectors = allConnectors.filter(
-    ({ metadata: { id, type } }) => existingConnectors.get(id)?.type !== type
+    ({ metadata: { id: name, platform } }) =>
```
Comment (simeng-li) [inline packages/core/src/connectors/index.ts:116]: should we align the property naming of id and name? As you use `connector.name` in DB but `metadata.id `in connector instances.

File: packages/core/src/connectors/index.ts [hunk 4]
```diff
@@ -108,11 +109,24 @@ export const getConnectorInstanceByType = async <T extends ConnectorInstance>(
...
+      existingConnectors.get([name, platform].join(' '))?.platform !== platform
   );
+
   await Promise.all(
-    newConnectors.map(async ({ metadata: { id, type } }) => insertConnector({ id, type }))
+    newConnectors.map(async ({ metadata }) => {
+      const id = await generateConnectorId();
+      await insertConnector({
+        id,
+        name: metadata.id,
```
Comment (simeng-li) [inline packages/core/src/connectors/index.ts:125]: same here, shoule we also align metadata.name?

File: packages/core/src/lib/connector.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      const id = connectorId();
+
+      if (!(await hasConnectorWithId(id))) {
+        return id;
+      }
+
+      throw new Error('Cannot generate connector ID in reasonable retries');
+    },
+    { retries, factor: 0 } // No need for exponential backoff
+  );
```
Comment (simeng-li) [inline packages/core/src/lib/connector.ts:20]: nit: looks like the same as `generateUserId` could extract using the same util method?  pass the findExistingId method as a param?

File: packages/connector-types/src/index.ts (hunk 1 above)
Comment (wangsijie) [reply packages/connector-types/src/index.ts:13]: I guess is for non-social connector, but then I think we can just make it "null"

Comment (wangsijie) [reply packages/connector-types/src/index.ts:13]: or optional

File: packages/core/src/lib/connector.ts (hunk 5 above)
Comment (wangsijie) [reply packages/core/src/lib/connector.ts:20]: Actually we can simply use `nanoid()`, `user` is a special case, you can refer to other tables.

File: packages/core/src/connectors/index.ts [hunk 6]
```diff
@@ -108,11 +109,24 @@ export const getConnectorInstanceByType = async <T extends ConnectorInstance>(
 
 export const initConnectors = async () => {
   const connectors = await findAllConnectors();
-  const existingConnectors = new Map(connectors.map((connector) => [connector.id, connector]));
+  const existingConnectors = new Map(
```
Comment (wangsijie) [inline packages/core/src/connectors/index.ts:112]: As for existing connectors, need to override metadata and other infos?

File: packages/core/src/lib/connector.ts (hunk 5 above)
Comment (IceHe) [reply packages/core/src/lib/connector.ts:20]: seems that there is no need to retry?

File: packages/phrases/src/index.ts [hunk 7]
```diff
@@ -2,19 +2,15 @@ import { NormalizeKeyPaths } from '@silverhand/essentials';
...
 import en from './locales/en';
 import zhCN from './locales/zh-cn';
-import { Resource } from './types';
+import { Resource, Language } from './types';
 
+export { Language } from './types';
 export type LogtoErrorCode = NormalizeKeyPaths<typeof en.errors>;
 export type LogtoErrorI18nKey = `errors:${LogtoErrorCode}`;
-export type Languages = keyof Resource;
+// Export type Languages = keyof Resource;
```
Comment (IceHe) [inline packages/phrases/src/index.ts:10]: should remove

File: .github/workflows/update-pr-metadata.yml [hunk 8]
```diff
@@ -80,12 +80,12 @@ jobs:
             const filesHeader = `|Name|Diff|\n|---|---|\n`;
             const filesData = result.map((row) => {
               const [diff, filename] = row.split('\t');
-              return `|${filename}|${format(Number(diff))}|\n`;
-            });
+              return `|${filename}|${format(Number(diff))}|`;
+            }).join('\n');
```
Comment (wangsijie) [inline .github/workflows/update-pr-metadata.yml:84]: do we need another PR for this?

File: packages/connector-alipay/src/index.ts [hunk 9]
```diff
@@ -104,7 +104,7 @@ export class AlipayConnector implements SocialConnector {
   };
 
   public getUserInfo: GetUserInfo = async (accessTokenObject) => {
-    const config = await this.getConfig(this.metadata.id);
+    const config = await this.getConfig(this.metadata.target, this.metadata.platform);
```
Comment (wangsijie) [inline packages/connector-alipay/src/index.ts:107]: Can we pass in "id"(in db) to this instance? Can create an issue to track.

File: packages/console/src/assets/docs/tutorial/integrate-sdk/android/index.mdx [hunk 10]
```diff
@@ -0,0 +1,285 @@
+import Tabs from '@theme/Tabs';
```
Comment (wangsijie) [inline packages/console/src/assets/docs/tutorial/integrate-sdk/android/index.mdx:1]: Not expected to seen in this PR?

File: packages/connector-alipay/src/index.ts (hunk 9 above)
Comment (darcyYe) [reply packages/connector-alipay/src/index.ts:107]: LOG-2397 was added to track this.

File: packages/ui/src/pages/Callback/index.tsx [hunk 11]
```diff
@@ -19,7 +19,9 @@ const Callback = () => {
   const { t } = useTranslation(undefined, { keyPrefix: 'main_flow' });
 
   const connectorLabel = useMemo(() => {
-    const connector = experienceSettings?.socialConnectors.find(({ id }) => id === connectorId);
+    const connector = experienceSettings?.socialConnectors.find(
+      ({ target }) => target === connectorId
```
Comment (wangsijie) [inline packages/ui/src/pages/Callback/index.tsx:23]: @simeng-li maybe refactor later? `connectorId === target` looks odd.

File: packages/core/src/connectors/index.test.ts [hunk 12]
```diff
@@ -131,15 +165,62 @@ describe('getConnectorInstances', () => {
...
+        code: 'entity.not_found',
+        target: 'invalid_target',
+        platfrom: ConnectorPlatform.Web,
+        status: 404,
+      })
+    );
+  });
+});
+
+describe('getConnectorInstanceByTargetAndPlatform', () => {
```
Comment (wangsijie) [inline packages/core/src/connectors/index.test.ts:196]: Can we remove this function?

**Review by wangsijie — APPROVED** [review]
